plugin/trino-iceberg/src/main/java/io/trino/plugin/iceberg/TypeConverter.java

```diff
@@ -117,7 +97,7 @@ private static org.apache.iceberg.types.Type toIcebergTypeInternal(Type type, Op
         if (type instanceof BooleanType) {
             return Types.BooleanType.get();
         }
-        if (type instanceof IntegerType) {
+        if (type instanceof IntegerType || type instanceof SmallintType || type instanceof TinyintType ) {
```

Comment:
I don't think we need to update here - The `getSupportedType` would perform an automatic coercion so that the type would still be an INTEGER at Iceberg side.